### ISSUE TITLE
Analysis of "global" statements doesn't believe in classes 

### ISSUE THREAD

**Issue by georgevdd** [PyCQA/pylint#3088]
A statement such as

>   global Xxx

results in GPyLint emitting a bogus invalid-name warning and also a bogus global-variable-undefined warning, if Xxx is a class.
A demo is attached.

[sad_lint.zip](https://github.com/PyCQA/pylint/files/3571660/sad_lint.zip)

**Comment by PCManticore** [comment]
For reference, here is the code in question:
```
from absl import app
from absl import flags

FLAGS = flags.FLAGS


NON_CLASS_VARIABLE = 3


class ClassToBeDecorated(object):
  pass


def decoration_that_depends_on_flags(cls):
  # In real life this would do something that depends on flags, so it can't be
  # applied using @syntax at class definition time.
  return cls


def apply_decoration():
  global ClassToBeDecorated  # Bogus lint warnings here.
  ClassToBeDecorated = decoration_that_depends_on_flags(ClassToBeDecorated)

  global NON_CLASS_VARIABLE  # No lint warnings here.
  NON_CLASS_VARIABLE = 4


def main(argv):
  del argv  # Unused
  apply_decoration()
  # Do the real main thing.

if __name__ == '__main__':
  app.run(main)

```

Please follow the recommended issue format when submitting issues, as it is simplifying the maintainer's life.

There are two issues here. First one is the `invalid-name`. I presume it's emitted because we expect all globals to be constants, rather than classes, and the default pattern for constants is upper case letters.
The second issue seems more important as it seems that the global analysis cannot find what the global declaration of `ClassToBeDecorated` is pointing at.
